doc/search-vulnerabilities-by-cpe-match-strings.org

```diff
@@ -0,0 +1,150 @@
+  may not be set to ~Not Applicable~ or ~ANY~.
+
+* Searching Vulnerabilities by CPE Match Strings
+  CTIA provides a way to query for vulnerabilities with configurations
+  that match combinations of CPE Match Strings.
+
+  The route is available through the endpoint ~GET
+  /ctia/vulnerability/cpe_match_strings~ and requires the query string
+  include an array of ~cpe23_match_strings~.
+
+  Many of the query string parameters documented in the endpoint [[file:search-metrics.org][~GET
+  /ctia/{entity-type}/search~]] are supported by ~GET
+  /ctia/vulnerability/cpe_match_strings~ with the following
+  exceptions:
+  - ~sort~
+  - ~offset~
+  - ~query~
+
+  A user can retrieve vulnerabilities applicable to various
+  combinations of hardware and software. The following example URL
+  encodes an array of the CPE match strings:
+  ~cpe:2.3:h:juniper:atp400:-:*:*:*:*:*:*:*~
+  ~cpe:2.3:o:juniper:advanced_threat_prevention:5.0.3:*:*:*:*:*:*:*~
+
+
+  #+BEGIN_SRC http
+    GET http://localhost:3000/ctia/vulnerability/cpe_match_strings?fields=title&cpe23_match_strings=cpe%3A2.3%3Ah%3Ajuniper%3Aatp400%3A*%3A*%3A*%3A*%3A*%3A*%3A*%3A*&cpe23_match_strings=cpe%3A2.3%3Ao%3Ajuniper%3Aadvanced_threat_prevention%3A5.0.3%3A*%3A*%3A*%3A*%3A*%3A*%3A*
+    Content-Type: application/json
+  #+END_SRC
+
+  Results:
+  #+BEGIN_SRC http
+  [
+    {
+      "groups": [
+        "Administrators"
+      ],
+      "tlp": "green",
+      "id": "http://localhost:3000/ctia/vulnerability/vulnerability-6b383506-48bb-4f20-bb0a-9844f9655efd",
+      "title": "CVE-2019-0021",
+      "owner": "Unknown"
+    }
+  ]
+  #+END_SRC
+
+  Response Headers:
+  #+BEGIN_SRC http
+   content-length: 186
+   content-type: application/json;charset=utf-8
+   date: Tue, 16 Mar 2021 19:14:30 GMT
+   etag: "87d3bcdea05c36e0c220de815c65d16b4676b65d"
+   x-content-type-options: nosniff
+   x-ctia-config: local
+   x-ctia-version: 10d96bac39a4c059be99b730e9d6ebbec81bcdb8
+   isue-4744-match-strings
+   x-ctim-version: 1.1.3
+   x-sort: ["vulnerability-6b383506-48bb-4f20-bb0a-9844f9655efd"]  x-total-hits: 1
+  #+END_SRC
+
+
+* Paging
+  Set the ~search_after~ query parameter to the value of the ~x-sort~
+  response header to retrieve the next page. The size of the page can
+  be set using the ~limit~ query parameter.  When ~limit~ is not
+  specified, the system default is used.
+
+  The endpoint utilizes two separate Elasticsearch queries. The first
+  query lazily retrieves up to ~limit~ Vulnerability IDs that match
+  the CPE Match Strings. The second query then utilizes up to ~limit~
+  Vulnerability IDs from the lazy seq to retreive and deliver the
+  Vulnerabilities to the end user. Both queries are sorted by ~id~ in
+  ascending order to support paging using the ~search_after~ query
+  parameter. Due to this design, ~offset~, ~query~, and ~sort~ can not
+  be supported.
+
+
+* Symbols Requiring Escapes
+
+
+  The following systems are utilized by the endpoint, and have
```

Comment:
I totally understand the complexity here. However I am not sure that this is user friendly.